filters/apiusagemonitoring/spec.go

```diff
@@ -30,50 +32,62 @@ func NewApiUsageMonitoring(
 	enabled bool,
 	realmKeys string,
 	clientKeys string,
-	defaultClientTrackingPattern string,
+	realmsTrackingPattern string,
 ) filters.Spec {
 	if !enabled {
-		log.Debugf("Filter %q is not enabled. Spec returns `noop` filters.", Name)
+		log.Debugf("filter %q is not enabled. spec returns `noop` filters.", Name)
 		return &noopSpec{&noopFilter{}}
 	}
 
-	// Parse realm keys comma separated list
+	// parse realm keys comma separated list
 	var realmKeyList []string
 	for _, key := range strings.Split(realmKeys, ",") {
 		strippedKey := strings.TrimSpace(key)
 		if strippedKey != "" {
 			realmKeyList = append(realmKeyList, strippedKey)
 		}
 	}
-	// Parse client keys comma separated list
+	// parse client keys comma separated list
 	var clientKeyList []string
 	for _, key := range strings.Split(clientKeys, ",") {
 		strippedKey := strings.TrimSpace(key)
 		if strippedKey != "" {
 			clientKeyList = append(clientKeyList, strippedKey)
 		}
 	}
+	// compile realms regex
+	realmsTrackingMatcher, err := regexp.Compile(realmsTrackingPattern)
+	if err != nil {
```

Comment:
i wonder if this approach can lead to a behavior not expected by the user. As I understand, we accept a pattern as a startup parameter, and if it's of invalid format, then we log it and use the default. My preference would be to abort the startup, and wondering if this approach was also considered.

Reply:
The constructor of the filter factory (`NewApiUsageMonitoring`) doesn't return an error. It's probably intended, so that Skipper doesn't fail in case a filter (factory) cannot be initialized.
I think it's OK and we also do provide a sensitive default.